x/mongo/driver/operation/hello.go

```diff
@@ -133,26 +472,15 @@ func (h *Hello) handshakeCommand(dst []byte, desc description.SelectedServer) ([
 	}
 	dst, _ = bsoncore.AppendArrayEnd(dst, idx)
 
-	// append client metadata
-	idx, dst = bsoncore.AppendDocumentElementStart(dst, "client")
-
-	didx, dst := bsoncore.AppendDocumentElementStart(dst, "driver")
-	dst = bsoncore.AppendStringElement(dst, "name", "mongo-go-driver")
-	dst = bsoncore.AppendStringElement(dst, "version", version.Driver)
-	dst, _ = bsoncore.AppendDocumentEnd(dst, didx)
-
-	didx, dst = bsoncore.AppendDocumentElementStart(dst, "os")
-	dst = bsoncore.AppendStringElement(dst, "type", runtime.GOOS)
-	dst = bsoncore.AppendStringElement(dst, "architecture", runtime.GOARCH)
-	dst, _ = bsoncore.AppendDocumentEnd(dst, didx)
+	clientMetadata, err := encodeClientMetadata(h.appname, maxClientMetadataSize)
+	if err != nil {
+		return dst, err
```

Comment:
Errors getting the client metadata shouldn't cause a handshake error. We should either ignore the error or log an error and continue with the handshake.